pkg/apis/core/v1alpha1/defaults_test.go

```diff
@@ -632,6 +638,32 @@ var _ = Describe("Defaults", func() {
 
 			Expect(obj.Spec.Kubernetes.EnableStaticTokenKubeconfig).To(PointTo(BeTrue()))
 		})
+
+		Context("k8s version < 1.25", func() {
+			BeforeEach(func() {
+				obj.Spec.Kubernetes.Version = "1.24.0"
+			})
+
+			Context("allowPrivilegedContainers field is not set", func() {
+				It("should set the field to true", func() {
+					SetDefaults_Shoot(obj)
+
+					Expect(obj.Spec.Kubernetes.AllowPrivilegedContainers).To(PointTo(BeTrue()))
+				})
+			})
+
+			Context("allowPrivilegedContainers field is set", func() {
+				BeforeEach(func() {
+					obj.Spec.Kubernetes.AllowPrivilegedContainers = pointer.Bool(false)
+				})
+
+				It("should not set the field", func() {
+					SetDefaults_Shoot(obj)
+
+					Expect(obj.Spec.Kubernetes.AllowPrivilegedContainers).To(PointTo(BeFalse()))
+				})
+			})
+		})
```

Comment:
What about adding a test for 1.25 version that it's not defaulted (same for `v1beta1`)?